fitbenchmarking/controllers/kmpfit_controller.py

```diff
@@ -0,0 +1,88 @@
+"""
+Implements a controller for the Kmpfit software.
+"""
+
+from kapteyn import kmpfit
+from fitbenchmarking.controllers.base_controller import Controller
+
+
+class KmpfitController(Controller):
+    """
+    Controller for Kmpfit
+    """
+
+    algorithm_check = {
+        'all': ['lm-lsqr'],
+        'ls': ['lm-lsqr'],
+        'deriv_free': [],
+        'general': [],
+        'simplex': [],
+        'trust_region': [],
+        'levenberg-marquardt': ['lm-lsqr'],
+        'gauss_newton': [],
+        'bfgs': [],
+        'conjugate_gradient': [],
+        'steepest_descent': [],
+        'global_optimization': []}
+
+    jacobian_enabled_solvers = ['lm-lsqr']
+
+    def __init__(self, cost_func):
+        """
+        Initialises variables used for temporary storage.
+        :param cost_func: Cost function object selected from options.
+        :type cost_func: subclass of
+                :class:`~fitbenchmarking.cost_func.base_cost_func.CostFunc`
+        """
+        super().__init__(cost_func)
+        self.result = None
+        self._status = None
+        self._popt = None
+        self.kmpfit_fitter = kmpfit.Fitter
+        self.kmpfit_object = None
+
+    def kmpfit_residuals(self, p, data):
+        """
+        Residuals for Kmpfit
+        """
+        _, _ = data
```

Comment:
I'm not sure what this does? Is it just to avoid the linter? I think  it's better to tell the linter not to check it or just pass the data to eval_r